fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -966,7 +966,7 @@ private ScoredPlan planSortOnly(@Nonnull CandidateScan candidateScan,
             }
         }
 
-        if (sort.isPrefixKey(index)) {
+        if (sort.isPrefixKey(index, true)) {
```

Comment:
Is there anywhere that calls `isPrefixKey` in a way that it doesn't expect them to be equivalent for sort?

Reply:
The other major use is checking for super-groups of grouped indexes (whose entries then need to be rolled up). That is somewhat sort-like; should `isPrefixKey` just always use the new `equals` then?

Reply:
Right, so if I'm understanding it right, the grouping value doesn't actually depend on whether the null stand in is `NULL_UNIQUE` or `NULL` as all nulls are collapsed there. I think maybe yeah, then, `isPrefixKey` should always use `equivalentForSort`. Or maybe instead of it being `equivalentForSort`, it's, like, `equivalentValues`, or something (indicating that the two expressions will always produce the same values for the same record)?

Reply:
I think I agree that the super-group checking is also "sort-like" so we could use `equivalentForSort()`. Notice, though, that `isPrefixKey()` (as currently written) needs something like "equivalent for sort ignoring children".  So maybe what we do is change the semantics of `equalsAtomic()` to match those of `equivalentForSort()`, and implement `equivalentForSort()` using `equalsAtomic()`, and also implement `equals()` using `equalsAtomic()`?